core/shared/src/main/scala/zio/prelude/ZSet.scala

```diff
@@ -86,6 +86,14 @@ final class ZSet[+A, +B] private (private val map: HashMap[A @uncheckedVariance,
       }
     })
 
+  def destruct[B1 >: B: Equal](implicit Inv: Inverse[Sum[B1]], Idn: Identity[Prod[B1]]): Option[(A, ZSet[A, B1])] =
```

Comment:
I don't think these operators are well defined. If the measure is an integer for example that could be negative I don't think returning one of the values and then a new set with the measure even more negative corresponds to the traditional definition of processing every element of a collection that you are doing for here.

Reply:
Makes sense, I've implemented it only for `*MultiSet`.